src/redis.c

```diff
@@ -353,6 +353,41 @@ static int redis_handle_query(redisContext *rh, redis_node_t *rn,
   return 0;
 } /* }}} int redis_handle_query */
 
+static int redis_db_stats(char *node, char const *info_line) /* {{{ */
+{
+  unsigned char db;
+  unsigned char max_db = 16;
+  static char field_name[10];
+  static char db_id[3];
+  char *str;
+  static char buf[MAX_REDIS_VAL_SIZE];
+  value_t val;
+  int i;
+
+  for (db = 0; db < max_db; db++) {
+	ssnprintf(field_name, sizeof(field_name), "db%d:keys", db);
+
+	str = strstr(info_line, field_name);
+	if (str) {
+	  str += strlen(field_name) + 1; /* also skip the '=' */
+	  for (i = 0; (*str && (isdigit((unsigned char)*str) || *str == '.')); i++, str++)
+		buf[i] = *str;
+	  buf[i] = '\0';
+
+	  if (parse_value(buf, &val, DS_TYPE_GAUGE) == -1) {
```

Comment:
Parse value may return other non-zero values on error, such as `EINVAL`, so you should check for `!= 0` instead.